ui/app/AppLayouts/Wallet/popups/swap/SwapModalAdaptor.qml

```diff
@@ -67,6 +102,39 @@ QObject {
         return null
     }
 
+    function fetchSuggestedRoutes() {
+        // Identify new swap with a different uuid
+        d.uuid = Utils.uuid()
+
+        let account = getSelectedAccount(root.swapFormData.selectedAccountIndex)
+        let accountAddress = account.address
+        let disabledChainIds = getDisabledChainIds(root.swapFormData.selectedNetworkChainId)
+        let preferedChainIds = getAllChainIds()
+
+        root.swapStore.fetchSuggestedRoutes(accountAddress, accountAddress, 
+            root.swapFormData.fromTokenAmount, root.swapFormData.fromTokensKey, root.swapFormData.toTokenKey,
+            disabledChainIds, disabledChainIds, preferedChainIds, 
+            Constants.SendType.Swap, "")
+    }
+
+    function sendApproveTx() {
+        let account = getSelectedAccount(root.swapFormData.selectedAccountIndex)
```

Comment:
The `swapFormData` should really have just the account address, not the selected index

Reply:
Makes sense to me, all other properties store the actual piece of info we need, not related to a model. @Khushboo-dev-cpp how do you feel about this?

Reply:
Case in point; we need to filter the balances (in the holding selector) according to the selected account:
```
/*,
                // TODO enable once AccountsModalHeader is reworked!!
                ValueFilter {
                    roleName: "account"
                    value: root.selectedSenderAccount.address
                }*/
```

and it's quite impractical to do using an index from the combo box

Reply:
yes I agree! we can change this

Reply:
Just a heads up, this is being changed to `property string selectedAccountAddress: ""` in https://github.com/status-im/status-desktop/pull/15099/files#r1631287128